packages/cozy-stack-client/src/DocumentCollection.js

```diff
@@ -308,6 +315,32 @@ The returned documents are paginated by the stack.
     }
   }
 
+  /**
+   * Returns a filtered list with all documents using a Mango selector,
+   * automatically fetching more documents if the total of documents is
+   * superior to the pagination limit.
+   * Can result in a lot of network requests.
+   *
+   The returned documents are paginated by the stack.
+   *
+   * @param {MangoPartialSelector} selector The Mango selector.
+   * @param {MangoQueryOptions} options MangoQueryOptions
+   * @returns {Promise<{data, skip, bookmark, next, execution_stats}>} Response in JSON
+   * @throws {FetchError}
+   */
+  async findAll(selector, options = {}) {
+    let next = true
+    let files = []
+    let bookmark = options.bookmark || undefined
+    while (next) {
+      const resp = await this.find(selector, { ...options, bookmark })
+      files.push(...resp.data)
+      bookmark = resp.bookmark
+      next = resp.next
+    }
```

Comment:
@paultranvan I'm wondering if we should not also manage the case were there is no bookmark? Does this case can still exist for stack-client? Or only for pouch-link?

Reply:
The stack transmits the CouchDB bookmark, which is always [there](https://docs.couchdb.org/en/stable/api/database/find.html), except when there is no doc :  https://github.com/cozy/cozy-stack/blob/095270c49a415aa10741400359b5cfdfc65c0c7d/pkg/couchdb/couchdb.go#L932
But in this case, `next` will be falsy

Reply:
OK so this case https://github.com/cozy/cozy-client/blob/master/packages/cozy-client/src/CozyClient.js#L994-L1003 is only there for pouch request, right?

Reply:
I assume, yes